docs/ONNXIFI.md

```diff
@@ -0,0 +1,52 @@
+# ONNX Interface for Framework Integration (ONNXIFI) Proposal
+
+We propose a cross-platform API for loading and executing ONNX graphs on optimized backends. High-level frameworks and applications can use this API to execute neural network and machine learning models. Hardware vendors can implement this API to expose specialized hardware accelerators and highly optimized software infrastructure to the users.
+
+## Core Features
+
+- Standardized interface for neural network inference on special-purpose accelerators, CPUs, GPUs, DSPs, and FPGAs
+- Based on widely supported technologies
+  - C API for function calls
+  - ONNX format for passing model graphs
+  - NCHW tensor layout for passing inputs and outputs
+- Dynamic discovery of available backends for model execution
+  - Multiple backends from different vendors can co-exist on the same system
+- Dynamic discovery of supported ONNX Operators on each backend
+
+### Optional Features:
+
+- Graphs with variable-shape inputs and/or outputs
+- Graphs with data-dependendent output shapes
+
+## How to Use ONNX Interface for Framework Integration
+
+0. (Optional) Use `onnxifi_library_load` to dynamically load the ONNX Interface for Framework Integration library.
+1. Call `onnxGetNumBackends` to get the number of available backends. Note that it can be 0.
+2. Call `onnxGetBackendInfo` to check additional information about any available backend.
+3. Call `onnxGetBackendCompatibility` to check if your model, or parts of it, can run on the backend.
+4. Call `onnxInitBackend` to initialize a backend, then call `onnxInitGraph` to offload one or more model graphs to the backend.
+5. Call `onnxSetGraphIO` to set inputs and output for the graph, then call `onnxRunGraph` to execute the graph(s). If your model works with fixed-size inputs, one call to `onnxSetGraphIO` is sufficient for multiple `onnxRunGraph` calls. For models with variable-size inputs, you'd need to call `onnxSetGraphIO` before each `onnxRunGraph` call.
+6. When done using the model, release the model graph(s) with `onnxReleaseGraph`, then release the backend with `onnxReleaseBackend`
+
+## How to Implement ONNX Interface for Framework Integration
+
+The minimum functionality an ONNXIFI implementation must provide is the following:
+
+- Support ONNX 1.0 model format.
+  - There is no minimum list of Operators a backed has to support.
```

Comment:
maybe we should put basic CNN here - but it's a separate discussion. @houseroad

Reply:
I'll leave it to Compliance Working Group.

Reply:
@dzhulgakov it's a great topic for the test and compliance working group. I have recorded it.

Reply:
backed -> backend